### ISSUE TITLE
choose header file

### ISSUE THREAD

**Issue by hidoudou** [libjpeg-turbo/libjpeg-turbo#219]
hi,i finished building libjpeg-turbo for Android on Mac OSX.
but one embarrassing thing is i donnot know to add which header file to my project.
i think it's a good idea to reorganize the source file, add include and src directory for example.

**Comment by dcommander** [comment]
libjpeg-turbo is a fork of a 27-year-old library, and its directory structure is inherited from that library.  I have no intention of changing it.  All of our API documentation is here:

https://libjpeg-turbo.org/Documentation/Documentation

The libjpeg API is documented in the various *.txt files.  I don't claim that that documentation is particularly user-friendly, because most of it is more than 20 years old as well.  libjpeg-turbo was originally intended to be a high-speed drop-in replacement for libjpeg, and thus its basic structure is heavily based on that of libjpeg.  I have added new features, including the fully documented TurboJPEG C and Java APIs, that make libjpeg-turbo much easier for first-time users to use than libjpeg, but the libjpeg portions of the code are legacy and not something I can change in a fundamental way.  If I moved the files around as you suggest, then people who are familiar with the libjpeg source tree would no longer be able to find them, and it would make it difficult for me to diff against IJG releases.
